src/stdlib/base.rs

```diff
@@ -79,4 +79,29 @@ pub fn load_base<'gc>(mc: MutationContext<'gc, '_>, root: LuaRoot<'gc>, env: Tab
         }),
     )
     .unwrap();
+
+    env.set(
+        mc,
+        String::new_static(b"type"),
+        Callback::new_immediate(mc, |args| {
+            if args.len() == 0 {
+                return Err(RuntimeError(Value::String(String::new_static(
+                    b"Missing argument to type",
+                )))
+                .into());
+            }
+            match args.get(0).cloned().unwrap() {
+                // Lua5.3 does not differentiate between a floating point and integer number
+                Value::Integer(_) => 
+                    Ok(CallbackResult::Return(vec![Value::String(
+                                    String::new_static(b"number"),
+                                    )])),
```

Comment:
I'm somewhat worried about this function in particular.  Is this a decent way to implement it considering that lua just throws away the integer vs number distinction?  Or should I try something else?

Reply:
I don't super like that you have to specify the string "number" twice, I wonder if type_name itself should be changed?

Reply:
Yeah, probably just change type_name to match, I think that's acceptable and simpler.